src/main/java/org/truffleruby/core/module/ModuleFields.java

```diff
@@ -301,19 +302,19 @@ public void prepend(RubyContext context, Node currentNode, DynamicObject module)
      * Set the value of a constant, possibly redefining it.
      */
     @TruffleBoundary
-    public void setConstant(RubyContext context, Node currentNode, String name, Object value) {
+    public boolean setConstant(RubyContext context, Node currentNode, String name, Object value) {
         if (context.getCoreLibrary().isLoadingRubyCore()) {
             final RubyConstant currentConstant = constants.get(name);
 
             if (currentConstant != null) {
-                return;
+                return false;
             }
         }
```

Comment:
Interesting, I don't remember why we ignore `setConstant()` during `isLoadingRubyCore`, I'll see if it's still needed.

Reply:
See #733

Reply:
Okay, I've rebased.